packages/filebrowser/src/listing.ts

```diff
@@ -425,30 +425,47 @@ export class DirListing extends Widget {
    *
    * @returns A promise that resolves when the operation is complete.
    */
-  async delete(): Promise<void> {
+  async delete(deleteToTrash: boolean = false): Promise<void> {
```

Comment:
Instead of adding a new argument here, could we duplicate the `const deleteToTrash = PageConfig.getOption('delete_to_trash') === 'true';` here?

Reply:
Did it in [8ad22be](https://github.com/jupyterlab/jupyterlab/pull/17359/commits/8ad22be99de659b90fabe814aaf030d80c3eb9cc).

Reply:
> Duplicate page config check to reduce public API changes
that makes sense, I wondered what's the motivation, but I completely forget that plugins may call this

Reply:
So motivation is two-fold:
- if a downstream subclassed this class and in a child method added an argument then technically it would be a breaking change
- if we want to add more arguments in the future we need to think about carefully as removing them is hard

Basically given the feedback on lab call it seemed likely that we will iterate on the delete behaviour in the future so I want to keep this PR as lightweight on the API choices as possible (and merge it ASAP as we are coming to a freeze on 4.4.0).

Reply:
I think calling `PageConfig` from `listings.ts` looks better, thanks @krassowski